usaspending_api/references/management/commands/fix_county_data.py

```diff
@@ -0,0 +1,395 @@
+                SELECT
+                    tn.place_of_performance_id,
+                    tf.place_of_perform_county_co AS county_code,
+                    tf.place_of_perform_county_na AS county_name
+                FROM transaction_normalized AS tn
+                LEFT OUTER JOIN transaction_fpds AS tf
+                    ON tf.transaction_id = tn.id
+                WHERE UPPER(tf.place_of_perform_country_c) = 'USA'
+            )
+            UPDATE references_location AS rl
+                SET county_code = tc.county_code,
+                    county_name = CASE WHEN rl.county_name IS NULL
+                                        THEN tc.county_name
+                                        ELSE rl.county_name END
+                FROM transaction_combined AS tc
+                WHERE tc.place_of_performance_id = rl.location_id
+                    AND rl.county_code IS NULL
+                    AND UPPER(rl.location_country_code) = 'USA'"""
+        )
+
+        logger.info("Finished FPDS PPOP location updates, finished location updates")
+
+    def add_arguments(self, parser):
+        parser.add_argument('-mv', '--matview', help='Create the matviews, make sure they do not already exist',
+                            action='store_true')
+        parser.add_argument('-dmv', '--delete_matview', help='Delete the matviews', action='store_true')
+        parser.add_argument('-fpdsle', '--fpds_le', help='Run FPDS Legal Entity updates', action='store_true')
+        parser.add_argument('-fpdsppop', '--fpds_ppop', help='Run FPDS PPOP updates', action='store_true')
+        parser.add_argument('-fabsle', '--fabs_le', help='Run FABS Legal Entity updates', action='store_true')
+        parser.add_argument('-fabsppop', '--fabs_ppop', help='Run FABS PPOP updates', action='store_true')
+        parser.add_argument('-l', '--location', help='Fix location county codes/names', action='store_true')
+        parser.add_argument('-a', '--all', help='Run all updates without creating or deleting matviews',
+                            action='store_true')
+        parser.add_argument('-am', '--all_matview', help='Run all updates and create and delete matviews',
+                            action='store_true')
+
+    def handle(self, *args, **options):
+        with connection.cursor() as sess:
+
+            logger.info("Starting county code fixes")
+
+            if options['all_matview'] or options['all']:
+                if options['all_matview']:
+                    self.create_matviews(sess)
+
+                self.update_fpds_le(sess)
+                self.update_fpds_ppop(sess)
+                self.update_fabs_le(sess)
+                self.update_fabs_ppop(sess)
+                self.update_location(sess)
+
+                if options['all_matview']:
+                    self.delete_matviews(sess)
+            else:
+                if options['matview']:
+                    self.create_matviews(sess)
```

Comment:
If we're picking a specific option here, would we want to run this matview creation _every_ time?

Reply:
Not really. If you create the matviews once and don't delete them, you can then run each section individually if you want (run just `fix_county_data -fpdsle` then wait until you have time before running `fix_county_data -fpdsppop` if needed). If you're starting from scratch you just have to call `fix_county_data -m -fpdsle` for example. It'll get to the matview first so no problems will be had.